### DISCUSSION TITLE
Early contributor observatons

### DISCUSSION THREAD

**Discussion by rimutaka** [shuttle-hq/shuttle#105] [General]
This is a list of little things that make no sense to an outsider. Purely FYI.

1. Is there a reason why Cargo.lock is included in the repo? If not, I suggest removing it

2. Examples refer to `shuttle-service = "0.2.4"` which may be few commits behind `main` branch so they fail to build. If `main` is always green it may be better to have `shuttle-service = {git = "https://github.com/rimutaka/shuttle.git"}` instead.

3. _e2e/README.md_ looks like a copy-paste from another project (`unveil-service`?). The document made no sense to me. I wanted to look up tests to understand how everything hangs together.

4. Does this _examples/rocket/hello-world_ example work in your dev env as a standalone project? I get an error.
```rust
extern crate rocket;
use rocket::{Build, Rocket};
#[macro_use]
extern crate shuttle_service;

#[get("/")]
fn index() -> &'static str {
    "Hello, world!"
}

fn rocket() -> Rocket<Build> {
    rocket::build().mount("/hello", routes![index]) // <---- compilation error because it can't find the macro
}

declare_service!(Rocket<Build>, rocket);
```
adding the macro name here 
```rust
use rocket::{Build, Rocket, routes};
```
makes it compile. What am I missing?

5. How do I go about building, debugging and testing my own app? Do I use the Docker container to simulate the environment? I could not find any instructions re. that.

**Comment by alsuren** [comment]
A few of these observations are similar to what I had written down when first investigating shuttle. I thought I would add them here for posterity: https://github.com/getsynth/shuttle/pull/108 (don't ask why I wrote them on a PR I regretted it immediately).
